### ISSUE TITLE
Emoji translation

### ISSUE THREAD

**Issue by MightyCreak** [vector-im/riot-web#11013]
**Is your suggestion related to a problem? Please describe.**
As a non native English user, I don't know all the English translation of each emojis. Sometime the French words (my mother tong) are easier to remember.

**Describe the solution you'd like**
On Slack, if it is set up in something else than English, when you enter emojis by text (i.e. starting with `:`), you can either use the English word or its translation (e.g. `:smile:` or its French translation `:sourire:`).

**Describe alternatives you've considered**
Only have the translation is not good either, I like the fact that you can use both English words and the ones from your language settings.

**Additional context**
Screenshots from Slack:
![image](https://user-images.githubusercontent.com/6382548/65962813-7cca4900-e427-11e9-8d6a-27659fc17a39.png)

![image](https://user-images.githubusercontent.com/6382548/65962828-89e73800-e427-11e9-887b-a56997d39e92.png)

**Comment by marceloaof** [comment]
Can't we at least translate the name of emojis? For example: Smile, instead of : smile :
That solution is applied to the verification request by emoji.

**Comment by strixaluco** [comment]
Emoji titles for verification are already translatable. You can contribute to your language translation here: https://translate.riot.im/

**Comment by marceloaof** [comment]
I was not clear enough, sorry. I meant to say that a subset of Emoji titles (for verification) are already translatable. Is it possible to expand that solution to solve partially the issue here?
: smile : would remain untranslated as : smile :
However, the title Smile would be translatable.
![image](https://user-images.githubusercontent.com/69255481/89922584-9f93c500-dbd5-11ea-96db-1650bbf46c76.png)

**Comment by strixaluco** [comment]
As far as I understand, all the titles/annotations translations of [emojibase](https://github.com/milesj/emojibase), which is used in Element, are supposed to be fetched from Unicode CLDR project [according to main developer](https://github.com/milesj/emojibase/issues/53#issuecomment-667478970). I might be missing something about terminology here because title translations don't appear in Element and I thought to look into it, but developer [stated](https://github.com/milesj/emojibase/issues/53#issuecomment-668143176) that it's an upstream's issue. Sorry if I confuse the readers.

**Comment by strixaluco** [comment]
Okay, so the problem does seem to be on Element side apparently. According to [emojibase docs](https://emojibase.dev/docs/datasets), each locale is stored in separate dataset, while matrix-react-sdk imports only English locale:
https://github.com/matrix-org/matrix-react-sdk/blob/be2d0c9de7edb498880147c327dc5035c9a7a62f/src/emoji.ts#L17

**Comment by strixaluco** [comment]
@MightyCreak meanwhile rework of emojibase shortcodes to enable localization has begun: https://github.com/milesj/emojibase/pull/64

**Comment by MightyCreak** [comment]
Just wanted to emphasize that Slack allows two translations at the same time: yours and English.

I think it is very useful since (as French is my mother tong), I often use French or English for the emoji completion. It depends on the context -- am I in a French or English room? but also on habits -- I am used to look for "thumbsup" all the time for instance.

**Comment by strixaluco** [comment]
There is a similar issue open: https://github.com/vector-im/element-web/issues/9298

**Comment by MayeulC** [comment]
> @MightyCreak meanwhile rework of emojibase shortcodes to enable localization has begun: [milesj/emojibase#64](https://github.com/milesj/emojibase/pull/64)

I just want to point out that this is now merged :)

**Comment by robintown** [comment]
https://emojibase.dev/docs/shortcodes#localization

To do this we'd have to switch back from IamCal shortcodes to the default Emojibase shortcodes, which might not be as much of an issue anymore, since in the past couple of years, Emojibase has updated their defaults to better match other platforms. It's a tradeoff.

**Comment by SimonBrandner** [comment]
> To do this we'd have to switch back from IamCal shortcodes to the default Emojibase shortcodes, which might not be as much of an issue anymore, since in the past couple of years, Emojibase has updated their defaults to better match other platforms. It's a tradeoff.

We only switched to `IamCal (Slack)` quite recently...

**Comment by Yajo** [comment]
If you could enable more than one style, it would be most accessible. It'd mean more paths to find the emoji, which is the goal after all.

**Comment by robintown** [comment]
@SimonBrandner Yes, but I'm saying that the issues solved by switching to IamCal could've maybe been solved just by updating Emojibase too, since we were using an outdated version at the time.

**Comment by SimonBrandner** [comment]
> @SimonBrandner Yes, but I'm saying that the issues solved by switching to IamCal could've maybe been solved just by updating Emojibase too, since we were using an outdated version at the time.

Ah, sorry, makes sense now, I've got https://github.com/matrix-org/matrix-react-sdk/pull/7286. I guess let's see after that is merged?

**Comment by tmarplatt** [comment]
So if I were to provide a suggestion or directly contribute emoji titles or shortcodes for my own language, where would I have to drop it?

**Comment by t3chguy** [comment]
@tmarplatt https://github.com/milesj/emojibase once we start applying i18n to emoji
